MultiShotCallerWithLights.ino

```diff
@@ -139,50 +85,116 @@ bool sampleAndCheckIfAboveThreshold(int inputPin, int threshold, int sampleCount
   for (int i = 0; i < sampleCount; i++){
     hitSampleSum = hitSampleSum + analogRead(inputPin);
   }
+  
+  if(DEBUG){
+    Serial.print("Hit reading Average: ");
+    Serial.println((hitSampleSum/sampleCount));
+  }
 
   // Check if average is above threshold
-  if ( (hitSampleSum/sampleCount) > threshold) {
+  if ( (hitSampleSum/sampleCount) > threshold) {    
     return true;
   }
 
   return false;
 }
 
-void setHitFlash(){
-  for(int i = 0; i < NUM_LEDS; i++){
-    leds[i] = CRGB::Red;
+void clearLeds(){
+  setLEDs("black", 0, NUM_LEDS, NULL);
+}
+
+void setLEDs(String color, int startLED, int endLED, int brightLevel){
+  int hue = 0;
+  int saturation = 255;
+  int brightness = LED_BRIGHTNESS;
+  if(brightLevel){
+    brightness = brightLevel;
+  }
+
+  if(color.equals("red")){
+    hue = 0;
+  } else if(color.equals("orange")){
+    hue = 22;
+  } else if(color.equals("green")){
+    hue = 85;
+  } else if(color.equals("black")){
+    brightness = 0;
+  }
+  
+  for(int i = startLED; i < endLED; i++){
+    leds[i].setHSV( hue, saturation, brightness);
   }
   FastLED.show();
-  prevHitPoints = 0;
 }
 
-void setHpIndicator(){
+void showHpIndicator(){
   if(prevHitPoints != currentHitPoints){
     double health_percent = double(currentHitPoints)/double(STARTING_HIT_POINTS);
     double led_count = double(NUM_LEDS/2);
     int greenLedCount = floor(health_percent * led_count);
-    if(DEBUG) {
-        Serial.print("health_precent:");
-        Serial.print(health_percent);
-        Serial.print(" lit led count: ");
-        Serial.println(greenLedCount);
-    }
+    
+    //clearLeds();
+    setLEDs("green", 0, NUM_LEDS, 1);
+    setLEDs("green", 0, greenLedCount, NULL);
+    setLEDs("green", NUM_LEDS/2, NUM_LEDS/2+greenLedCount, NULL);
+    prevHitPoints = currentHitPoints;
+  }
+}
+
+void beep(int duration){
+  digitalWrite(LED_BUILTIN, HIGH);
+  delay(duration);
+  digitalWrite(LED_BUILTIN, LOW);
```

Comment:
you might be getting wonky beepy race condition behavior by doing this - the beep pin is tied to the `LED_BUILTIN` so on line 179 when you set this HIGH and then LOW in the condition this could be fighting it

Reply:
Good point. Luckily, this function is only being called on startup and when dead, so chances are low (I think), and I haven't seen wonkiness yet.